SeaSharpe_CVGS/SeaSharpe_CVGS/Controllers/OrderController.cs

```diff
@@ -302,6 +302,8 @@ public ActionResult Cart(int? id)
             //validate that the member has a cart
             var exists = db.Orders.Where(m => m.Member.Id == member.Id).Where(d => d.OrderPlacementDate == null).Any();
 
+            TempData["hasAddress"] = db.Addresses.Where(a => a.Member.Id == member.Id).FirstOrDefault();

```

Comment:
TempData isn't necessary here, could have been ViewData (ViewBag). [This gives a good explanation of the difference] (http://www.dotnet-tricks.com/Tutorial/mvc/9KHW190712-ViewData-vs-ViewBag-vs-TempData-vs-Session.html)